app/services/system-alert.js

```diff
@@ -0,0 +1,68 @@
+import Service, { inject as service } from '@ember/service';
+import { A } from '@ember/array';
+import moment from 'moment';
+import { tracked } from '@glimmer/tracking';
+import { later } from '@ember/runloop';
+import { action } from '@ember/object';
+
+export default class SystemAlertService extends Service {
+  @service store;
+
+  @tracked alerts = A([]);
+  updateInterval = 60 * 1000;
+
+  get currentAlerts () {
+    const now = new Date();
+    return A(this.alerts.filter(a => a.beginDate < now && a.endDate > now));
+  }
+
+  get futureAlerts () {
+    const now = new Date();
+    return A(this.alerts.filter(a => a.beginDate > now));
+  }
+
+  get unconfirmedAlerts () {
+    return A(this.currentAlerts.filter((a) => !a.confirmed));
+  }
+
+  constructor () {
+    super(...arguments);
+    this.lifecycle();
+  }
+
+  async lifecycle () {
+    try {
+      await this.updateAlerts();
+    }
+    finally {
+      later(this, this.lifecycle, this.updateInterval);
+    }
+  }
+
+  @action
+  confirmAlert (alert) {
+    alert.set('confirmed', true);
+  }
+
+  async updateAlerts() {
+    /*
+     * Below query fetches all alerts with end date greater then start of the day.
+     * This makes for only 1 unique request per day, which is good for request caching.
+     */
+    const today = moment().startOf('day').format();
```

Comment:
Here we use moment again? Hurray, but why is this mixed. I'm dying to find out! 😄

Reply:
As said above, entirely no reason why you couldn't use moment (getting the `startOf` done in vanilla JS would be much less readable).
Looking at it, the `format` doesn't look very consequent with what I'm preeching though :wink: ... that should be `toDate` :ok_hand:

Reply:
Honestly I'd always use `moment` for operations on Date. Doing that vanilla is just a pain really. I think we can leave this out of the discussion as this shouldn't change imo. Except for `toDate`. The thing is that I really would not prefer a reliance on the Date object as a default.